version/cmd/version/version.go

```diff
@@ -56,34 +40,54 @@ func main() {
 		flag.PrintDefaults()
 		os.Exit(1)
 	}
-	result, err := fn(tag)
+	result, err := fn(tag, bt)
 	if err != nil {
 		log.Fatal(err)
 	}
 	fmt.Println(result)
 }
 
-func getTag() (string, error) {
+// buildType getter func that matches the signature of other version funcs
+func buildType(_ string, bt version.BuildType) (string, error) {
+	return string(bt), nil
+}
+
+// implements the version.BuildEnv interface using the real build environment
+type BuildEnv struct{}
+
+// Returns true if we are building from a CI (rather than a local or dev build)
+func (b *BuildEnv) IsCI() bool {
```

Comment:
You don't actually need a pointer here, or even a receiver name, since these methods only exist to satisfy an interface.

You could have
```
func (BuildEnv) IsCI() bool {}
```
for instance.